keps/sig-instrumentation/4785-resource-state-metrics/README.md

```diff
@@ -0,0 +1,1170 @@
+      reason: EventHandlerSucceeded                                                                                               
+      status: "True                                                                                                           
+      type: Processed
+ ```
+- [x] Other: `http_request_duration_seconds` is a histogram metric exposed on 
+  the telemetry port which is useful for observing the trends in requests for
+  the generated metrics.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+TBD.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [x] Metrics: Telemetry metrics exposed on `RSM_SELF_PORT`.
+- [x] Other: `healthz`, `readyz`, and `livez` endpoints are exposed on
+  `RSM_SELF_PORT`, `RSM_MAIN_PORT`, and `RSM_SELF_PORT`,
+  respectively.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+TBD.
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+The controller __directly__ imports the following `k8s.io` packages:
+* `k8s.io/api`
+* `k8s.io/apimachinery`
+* `k8s.io/client-go`
+* `k8s.io/code-generator`
+* `k8s.io/klog/v2`
+* `k8s.io/utils`
+
+###### Does this feature depend on any specific services running in the cluster?
+
+<!--
+Think about both cluster-level services (e.g. metrics-server) as well
+as node-level agents (e.g. specific version of CRI). Focus on external or
+optional services that are needed. For example, if this feature depends on
+a cloud provider API, or upon an external software-defined storage or network
+control plane.
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+The controller relies on core Kubernetes components.
```

Comment:
Be more specific - it's effectively serving stack only, right (kube-apiserver + etcd + potentially webhooks etc.)